### DISCUSSION TITLE
Analog controls

### DISCUSSION THREAD

**Discussion by MossCollector** [fhdm-dev/scoppy#59] [Scoppy Front end designs]
Hi everyone.
First i should probably clarify i have little to no knowledge about coding.
I was wondering is it possible to use Pico GPIOs to switch options like channel selection, horizontal and vertical position etc. with buttons and rotary encoders. 
My intention is to have semi-analog interface to permamently mounted encased old phone with most important controls in form of knobs and switches.

Thanks in advance :)

**Comment by fhdm-dev** [comment]
Hi,

The firmware and app currently don't support this but I think it would be awesome. Do you know the exact number of buttons and encoders you would be using? I'm just wondering if the Pico has enough GPIOs (especially if you want to keep the Logic Analyzer and multiple voltage range functionality) or whether some sort of IO expender would be needed.

**Comment by MossCollector** [comment]
Hi, 
So i went through the documentation, and it seems that on base board there are 9 free GPIOs left (GPIO14-21 and GPIO28), so inspired by your suggestion i started reading about IO expanders and they seems the way to go. Here where i live MCP23008s are the only one avaliable in reasonable price. 
I do not know how hard is to implement such amount of analog controls, but i transcribed almost every one possible (that would make sense in my opinion) and here is google spreadsheet. Two MCP23008 would be enough, with three every control would be covered. 

https://docs.google.com/spreadsheets/d/1ekSfudfNgP11_UqEOyGxvKViuh04o8TD3qlBLl34rB4/edit#gid=0

**Reply by fhdm-dev** [reply]
I've never used an MCP23008 but I don't think it would be particularly difficult to implement, just a bit time consuming because of the large number of controls.
I'll get a few MCP23008s the next time I place an order with Mouser and see what I can do.

**Reply by MossCollector** [reply]
If there is another way to do this, maybe with another IC, which you know well or there is a smarter way to implement such controls, please do what's best in your opinion :). 
I'm already grateful for your fast answers and willingness to work on this idea.

**Reply by fhdm-dev** [reply]
Hmmm. An MCU rather than a dedicated I/O expander might also work and probably wouldn't cost any more eg. something like an AVR32DB28. 

However, I'm probably getting ahead of myself at bit. I might just try a minimal implementation first using the Pico GPIOs. I'm not sure what the resposiveness will be like because I'm asking a lot of the RP2040 already!

This won't be in the next release but hopefully it won't be too far away :)

**Comment by endarthur** [comment]
Hi! I'm not sure if the right thing would be to message here or create a new topic, but would it maybe be possible to support just adding a couple (or 3, if we really have 9 remaining gpio pins) of rotary encoders and maybe setting on the app what would they do?

By the way, thanks for the great project!

**Reply by fhdm-dev** [reply]
That certainly would be simpler than adding on an IO expander (like I suggested in an earlier post on this thread). Some of the code has already been written so hopefully the firmware and app will have some rudamentary functionality for analog controls before too long....tho' I said something similar 2 years ago :smile:
